<launch>
	<arg name="sim" default="false"/>
	
	<node unless="$(arg sim)" name="lidar" pkg="urg_node" type="urg_node" args="_serial_port:=/dev/serial/by-id/usb-Hokuyo_Data_Flex_for_USB_URG-Series_USB_Driver-if00">
		<remap from="scan" to="scan_raw"/>
		<param name="angle_min" value="-1.55"/>
		<param name="angle_max" value="1.57"/>
	</node>

	<node name="lidar_mux" pkg="topic_tools" type="mux" args="scan scan_raw scan_filtered"/>

	<node name="can_filter" pkg="laser_filters" type="scan_to_scan_filter_chain">
		<rosparam command="load" file="$(find alpha_sensors)/config/can_filter_config.yaml"/>
		<remap from="scan" to="scan_raw"/>
	</node>

	<!-- CAMERA FOV FILTER -->
	<node name="camera_fov_filter" pkg="laser_filters" type="scan_to_scan_filter_chain">
		<rosparam command="load" file="$(find alpha_sensors)/config/camera_fov_filter.yaml"/>
		<remap from="scan" to="scan_raw"/>
		<remap from="scan_filtered" to="camera_scan"/>
	</node>

</launch>
